src/gt4py/_core/locking.py

```diff
@@ -0,0 +1,36 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import pathlib
+from typing import ContextManager, Protocol
+
+import filelock
+
+
+class _LockConstructor(Protocol):
+    def __call__(self, path: str, /) -> ContextManager: ...
+
+
+# Changing the lock is private API, but could be made public in the future
+# The second element (str) is an identifier of the lock implementation to avoid conflicts
+# when switching between different lock implementations.
+_lock: tuple[_LockConstructor, str] = (filelock.FileLock, "filelock")
```

Comment:
I don't think the custom identifier is really needed. An identifier can be easily computed from the lockConstructor class name. For example using the "abbreviation" (a10n) from eve utils (or directly the original implementation from boltons: https://boltons.readthedocs.io/en/latest/strutils.html#boltons.strutils.a10n)
```python
identifier = f"{eve.utils.a10n(_lock.__module__)}_{eve.utils.a10n(_lock.__class__.__name__)}"
```

```

Reply:
thanks, I wanted to something like this, but didn't know which hash to use.